### PR TITLE
Cache Storage: Make test setup resilient against buggy implementations

### PR DESCRIPTION
**Pull request by inexorabletash**
One test asserts that the cache name "unpaired\uD800" does not get mapped to "unpaired\uFFFD", but relies on a cache by that name not existing. If an implementation is buggy it will implicitly create the cache, causing the test to fail the setup checks on subsequent runs, as opposed to hitting the actual test assertions.

Change the test to delete the caches so that it starts with a clean slate regardless of whether or it passes the test.

### CODE REVIEW

**Comment by inexorabletash** [comment]
@wanderview - can you take a peek?

File: service-workers/cache-storage/script-tests/cache-storage.js [hunk 1]
```diff
@@ -170,8 +170,15 @@ promise_test(function(t) {
 
 promise_test(function(t) {
     var bad_name = 'unpaired\uD800';
-    var converted_name = 'unpaired\uFFFD'; // Don't create cache with this name.
-    return self.caches.has(converted_name)
+    var converted_name = 'unpaired\uFFFD';
+
+    return Promise.all([
+          self.caches.delete(bad_name),
+          self.caches.delete(converted_name)
```
Comment (wanderview) [inline service-workers/cache-storage/script-tests/cache-storage.js]: I'm confused.  Why is this necessary?

Comment (inexorabletash) [inline service-workers/cache-storage/script-tests/cache-storage.js]: Quoting from above:

> One test asserts that the cache name "unpaired\uD800" does not get mapped to "unpaired\uFFFD", but relies on a cache by that name not existing. If an implementation is buggy it will implicitly create the cache, causing the test to fail the setup checks on subsequent runs, as opposed to hitting the actual test assertions.
>
> Change the test to delete the caches so that it starts with a clean slate regardless of whether or it passes the test.

Comment (wanderview) [inline service-workers/cache-storage/script-tests/cache-storage.js]: Can you just add a clarifying comment to the test?  Looking at it on its own suggests that it should be deleting something, but its really there as a failsafe.  It might be a bit clearer if you just iterated cache.keys() and deleted all cache objects to get a blank slate.

**Review by wanderview — APPROVED** [review]

**Comment by inexorabletash** [comment]
Dang it, sorry, rebased. :P Let's see if I can make this not suck....

**Comment by inexorabletash** [comment]
@wanderview - can you take another quick look?

(I also renamed the string variable per review of Blink's copy of the test)

**Review by wanderview — APPROVED** [review]
